Image not loading on Android app

### Expected behavior
The image should be visible when it loads all blogs in the list
The site support only HTTP, not HTTPS (HTTPS is not forced) and when the Android app load it is showed a blank image.

<img width="606" alt="Screen Shot 2019-05-25 at 7 58 24 PM" src="https://user-images.githubusercontent.com/1651643/58375699-d1b0fc80-7f27-11e9-87e6-bcade004625e.png">

### Actual behavior
Blank image on loading. not loading.

### Steps to reproduce the behavior
Open wordpress app -> add custom blog url -> Login with creds -> Blog posts 


##### Tested on [device], Android [version], WPAndroid [version]
Android 6.0 and above.

##### Issue
Android logs say
`java.io.FileNotFoundException(No content provider: https://brightersight.ca/eds-blog/wp-content/uploads/2019/05/640px-Flag_of_the_Canadian_Forces.svg_.png?w=1032&h=660)`

The problem is "It's trying to load image from HTTPS but as I mentioned earlier. our website running without ssl. How come wordpress android app force images to load from https even our domain itself is running from http ??????

our blog is at https://brightersight.ca/eds-blog

original image url is, without https
http://brightersight.ca/eds-blog/wp-content/uploads/2019/05/640px-Flag_of_the_Canadian_Forces.svg_.png?w=1032&h=660

It's the same still trying to load it from https.

`05-30 22:18:23.526 8996-8996/? W/Glide: Load failed for https://brightersight.ca/eds-blog/wp-content/uploads/2019/05/comic-characters-2024772_960_720-1.jpg?w=1032&h=660 with size [984x294]
    class com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There were 2 causes:
    com.android.volley.NoConnectionError(javax.net.ssl.SSLPeerUnverifiedException: Hostname brightersight.ca not verified:
        certificate: sha256/2Emgbdya/FmPZwyqzmXQJWGWjrFj8+TP32kthTY9ViY=
        DN: CN=*.rteng.pro,OU=PositiveSSL Wildcard,OU=Domain Control Validated
        subjectAltNames: [*.rteng.pro, rteng.pro])
    java.io.FileNotFoundException(No content provider: https://brightersight.ca/eds-blog/wp-content/uploads/2019/05/comic-characters-2024772_960_720-1.jpg?w=1032&h=660)
     call GlideException#logRootCauses(String) for more detail
      Cause (1 of 2): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, REMOTE
    There was 1 cause:
    com.android.volley.NoConnectionError(javax.net.ssl.SSLPeerUnverifiedException: Hostname brightersight.ca not verified:
        certificate: sha256/2Emgbdya/FmPZwyqzmXQJWGWjrFj8+TP32kthTY9ViY=
        DN: CN=*.rteng.pro,OU=PositiveSSL Wildcard,OU=Domain Control Validated
        subjectAltNames: [*.rteng.pro, rteng.pro])
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetch failed
    There was 1 cause:
    com.android.volley.NoConnectionError(javax.net.ssl.SSLPeerUnverifiedException: Hostname brightersight.ca not verified:
        certificate: sha256/2Emgbdya/FmPZwyqzmXQJWGWjrFj8+TP32kthTY9ViY=
        DN: CN=*.rteng.pro,OU=PositiveSSL Wildcard,OU=Domain Control Validated
        subjectAltNames: [*.rteng.pro, rteng.pro])
     call GlideException#logRootCauses(String) for more detail
          Cause (1 of 1): class com.android.volley.NoConnectionError: javax.net.ssl.SSLPeerUnverifiedException: Hostname brightersight.ca not verified:
        certificate: sha256/2Emgbdya/FmPZwyqzmXQJWGWjrFj8+TP32kthTY9ViY=
        DN: CN=*.rteng.pro,OU=PositiveSSL Wildcard,OU=Domain Control Validated
        subjectAltNames: [*.rteng.pro, rteng.pro]
      Cause (2 of 2): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class android.content.res.AssetFileDescriptor, LOCAL
    There was 1 cause:
    java.io.FileNotFoundException(No content provider: https://brightersight.ca/eds-blog/wp-content/uploads/2019/05/comic-characters-2024772_960_720-1.jpg?w=1032&h=660)
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class java.io.FileNotFoundException: No content provider: https://brightersight.ca/eds-blog/wp-content/uploads/2019/05/comic-characters-2024772_960_720-1.jpg?w=1032&h=660
05-30 22:18:23.536 8996-8996/? I/Glide: Root cause (1 of 2)
    com.android.volley.NoConnectionError: javax.net.ssl.SSLPeerUnverifiedException: Hostname brightersight.ca not verified:
        certificate: sha256/2Emgbdya/FmPZwyqzmXQJWGWjrFj8+TP32kthTY9ViY=
        DN: CN=*.rteng.pro,OU=PositiveSSL Wildcard,OU=Domain Control Validated
        subjectAltNames: [*.rteng.pro, rteng.pro]
        at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:158)
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:114)
     Caused by: javax.net.ssl.SSLPeerUnverifiedException: Hostname brightersight.ca not verified:
        certificate: sha256/2Emgbdya/FmPZwyqzmXQJWGWjrFj8+TP32kthTY9ViY=
        DN: CN=*.rteng.pro,OU=PositiveSSL Wildcard,OU=Domain Control Validated
        subjectAltNames: [*.rteng.pro, rteng.pro]
        at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.java:308)
        at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.java:268)
        at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:160)
        at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:256)
        at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:134)
        at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:113)
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
        at okhttp3.RealCall.execute(RealCall.java:77)
        at org.wordpress.android.fluxc.network.OkHttpStack.performRequest(OkHttpStack.java:66)
        at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:97)
        	... 1 more
05-30 22:18:23.536 8996-8996/? I/Glide: Root cause (2 of 2)
    java.io.FileNotFoundException: No content provider: https://brightersight.ca/eds-blog/wp-content/uploads/2019/05/comic-characters-2024772_960_720-1.jpg?w=1032&h=660
        at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1092)
        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:944)
        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:867)
        at com.bumptech.glide.load.data.AssetFileDescriptorLocalUriFetcher.loadResource(AssetFileDescriptorLocalUriFetcher.java:22)
        at com.bumptech.glide.load.data.AssetFileDescriptorLocalUriFetcher.loadResource(AssetFileDescriptorLocalUriFetcher.java:13)
        at com.bumptech.glide.load.data.LocalUriFetcher.loadData(LocalUriFetcher.java:44)
        at com.bumptech.glide.load.engine.SourceGenerator.startNext(SourceGenerator.java:62)
        at com.bumptech.glide.load.engine.DecodeJob.runGenerators(DecodeJob.java:299)
        at com.bumptech.glide.load.engine.DecodeJob.runWrapped(DecodeJob.java:269)
        at com.bumptech.glide.load.engine.DecodeJob.run(DecodeJob.java:230)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
        at java.lang.Thread.run(Thread.java:818)
        at com.bumptech.glide.load.engine.executor.GlideExecutor$DefaultThreadFactory$1.run(GlideExecutor.java:446)`

--- a/RELEASE-NOTES.txt
+++ b/RELEASE-NOTES.txt
@@ -1,6 +1,7 @@
 16.2
 -----
- 
+* [*] Posts List: fixed bug that prevented showing the Featured Image of a post for pure self-hosted sites [https://github.com/wordpress-mobile/WordPress-Android/pull/13323]
+
 16.1
 -----
 * [***] Block Editor: Adds new option to select from a variety of predefined page templates when creating a new page for a Gutenberg site.

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/FeaturedImageHelper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/FeaturedImageHelper.kt
@@ -169,8 +169,7 @@ class FeaturedImageHelper @Inject constructor(
                 mediaUri,
                 maxDimen,
                 maxDimen,
-                !siteUtilsWrapper.isPhotonCapable(site),
-                site.isPrivateWPComAtomic
+                siteUtilsWrapper.getAccessibilityInfoFromSite(site)
         )
         return FeaturedImageData(FeaturedImageState.REMOTE_IMAGE_LOADING, photonUrl)
     }

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java
@@ -70,9 +70,20 @@ public class ReaderUtils {
         final String unescapedUrl = StringEscapeUtils.unescapeHtml4(imageUrl);
 
         if (siteAccessibilityInfo.isPhotonCapable()) {
-            return PhotonUtils.getPhotonImageUrl(unescapedUrl, width, height, quality, siteAccessibilityInfo.getSiteVisibility() == SiteVisibility.PRIVATE_ATOMIC);
+            return PhotonUtils.getPhotonImageUrl(
+                    unescapedUrl,
+                    width,
+                    height,
+                    quality,
+                    siteAccessibilityInfo.getSiteVisibility() == SiteVisibility.PRIVATE_ATOMIC
+            );
         } else {
-            return getImageForDisplayWithoutPhoton(unescapedUrl, width, height, siteAccessibilityInfo.getSiteVisibility() == SiteVisibility.PRIVATE);
+            return getImageForDisplayWithoutPhoton(
+                    unescapedUrl,
+                    width,
+                    height,
+                    siteAccessibilityInfo.getSiteVisibility() == SiteVisibility.PRIVATE
+            );
         }
     }
 
@@ -81,7 +92,12 @@ public class ReaderUtils {
      * (i.e. a private post - images in private posts can't use photon
      * but these are usually wp images so they support the h= and w= query params)
      */
-    private static String getImageForDisplayWithoutPhoton(final String imageUrl, int width, int height, boolean forceHttps) {
+    private static String getImageForDisplayWithoutPhoton(
+            final String imageUrl,
+            int width,
+            int height,
+            boolean forceHttps
+    ) {
         if (TextUtils.isEmpty(imageUrl)) {
             return "";
         }

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java
@@ -49,17 +49,39 @@ public class ReaderUtils {
 
 
         if (isPrivate && !isPrivateAtomic) {
-            return getPrivateImageForDisplay(unescapedUrl, width, height);
+            return getImageForDisplayWithoutPhoton(unescapedUrl, width, height, true);
         } else {
             return PhotonUtils.getPhotonImageUrl(unescapedUrl, width, height, quality, isPrivateAtomic);
         }
     }
 
+    public static String getResizedImageUrl(final String imageUrl,
+                                            int width,
+                                            int height,
+                                            SiteAccessibilityInfo siteAccessibilityInfo) {
+        return getResizedImageUrl(imageUrl, width, height, siteAccessibilityInfo, PhotonUtils.Quality.MEDIUM);
+    }
+
+    public static String getResizedImageUrl(final String imageUrl,
+                                            int width,
+                                            int height,
+                                            SiteAccessibilityInfo siteAccessibilityInfo,
+                                            PhotonUtils.Quality quality) {
+        final String unescapedUrl = StringEscapeUtils.unescapeHtml4(imageUrl);
+
+        if (siteAccessibilityInfo.isPhotonCapable()) {
+            return PhotonUtils.getPhotonImageUrl(unescapedUrl, width, height, quality, siteAccessibilityInfo.getSiteVisibility() == SiteVisibility.PRIVATE_ATOMIC);
+        } else {
+            return getImageForDisplayWithoutPhoton(unescapedUrl, width, height, siteAccessibilityInfo.getSiteVisibility() == SiteVisibility.PRIVATE);
+        }
+    }
+
     /*
-     * use this to request a reduced size image from a private post - images in private posts can't
-     * use photon but these are usually wp images so they support the h= and w= query params
+     * use this to request a reduced size image from not photon capable sites
+     * (i.e. a private post - images in private posts can't use photon
+     * but these are usually wp images so they support the h= and w= query params)
      */
-    private static String getPrivateImageForDisplay(final String imageUrl, int width, int height) {
+    private static String getImageForDisplayWithoutPhoton(final String imageUrl, int width, int height, boolean forceHttps) {
         if (TextUtils.isEmpty(imageUrl)) {
             return "";
         }
@@ -74,8 +96,14 @@ public class ReaderUtils {
         } else {
             query = "";
         }
-        // remove the existing query string, add the new one, and make sure the url is https:
-        return UrlUtils.removeQuery(UrlUtils.makeHttps(imageUrl)) + query;
+
+        if (forceHttps) {
+            // remove the existing query string, add the new one, and make sure the url is https:
+            return UrlUtils.removeQuery(UrlUtils.makeHttps(imageUrl)) + query;
+        } else {
+            // remove the existing query string, add the new one
+            return UrlUtils.removeQuery(imageUrl) + query;
+        }
     }
 
     /*

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtilsWrapper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtilsWrapper.kt
@@ -22,8 +22,12 @@ class ReaderUtilsWrapper @Inject constructor(
     fun getResizedImageUrl(imageUrl: String?, width: Int, height: Int, isPrivate: Boolean, isAtomic: Boolean): String? =
             ReaderUtils.getResizedImageUrl(imageUrl, width, height, isPrivate, isAtomic)
 
-    fun getResizedImageUrl(imageUrl: String?, width: Int, height: Int, siteAccessibilityInfo: SiteAccessibilityInfo): String? =
-            ReaderUtils.getResizedImageUrl(imageUrl, width, height, siteAccessibilityInfo)
+    fun getResizedImageUrl(
+        imageUrl: String?,
+        width: Int,
+        height: Int,
+        siteAccessibilityInfo: SiteAccessibilityInfo
+    ): String? = ReaderUtils.getResizedImageUrl(imageUrl, width, height, siteAccessibilityInfo)
 
     fun getTagFromTagName(tagName: String, tagType: ReaderTagType): ReaderTag =
             ReaderUtils.getTagFromTagName(tagName, tagType)

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtilsWrapper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtilsWrapper.kt
@@ -22,6 +22,9 @@ class ReaderUtilsWrapper @Inject constructor(
     fun getResizedImageUrl(imageUrl: String?, width: Int, height: Int, isPrivate: Boolean, isAtomic: Boolean): String? =
             ReaderUtils.getResizedImageUrl(imageUrl, width, height, isPrivate, isAtomic)
 
+    fun getResizedImageUrl(imageUrl: String?, width: Int, height: Int, siteAccessibilityInfo: SiteAccessibilityInfo): String? =
+            ReaderUtils.getResizedImageUrl(imageUrl, width, height, siteAccessibilityInfo)
+
     fun getTagFromTagName(tagName: String, tagType: ReaderTagType): ReaderTag =
             ReaderUtils.getTagFromTagName(tagName, tagType)
 

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/SiteAccessibilityInfo.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/SiteAccessibilityInfo.kt
@@ -1,0 +1,9 @@
+package org.wordpress.android.ui.reader.utils
+
+enum class SiteVisibility {
+    PRIVATE,
+    PRIVATE_ATOMIC,
+    PUBLIC
+}
+
+data class SiteAccessibilityInfo(val siteVisibility: SiteVisibility, val isPhotonCapable: Boolean)

--- a/WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java
+++ b/WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java
@@ -5,6 +5,7 @@ import android.text.TextUtils;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
+import org.jetbrains.annotations.NotNull;
 import org.wordpress.android.WordPress;
 import org.wordpress.android.analytics.AnalyticsTracker.Stat;
 import org.wordpress.android.fluxc.Dispatcher;
@@ -16,6 +17,8 @@ import org.wordpress.android.fluxc.store.SiteStore.DesignateMobileEditorForAllSi
 import org.wordpress.android.fluxc.store.SiteStore.DesignateMobileEditorPayload;
 import org.wordpress.android.ui.plans.PlansConstants;
 import org.wordpress.android.ui.prefs.AppPrefs;
+import org.wordpress.android.ui.reader.utils.SiteAccessibilityInfo;
+import org.wordpress.android.ui.reader.utils.SiteVisibility;
 import org.wordpress.android.util.analytics.AnalyticsUtils;
 import org.wordpress.android.util.analytics.AnalyticsUtils.BlockEditorEnabledSource;
 import org.wordpress.android.util.helpers.Version;
@@ -244,6 +247,20 @@ public class SiteUtils {
     public static String getSiteIconUrl(SiteModel site, int size) {
         return PhotonUtils.getPhotonImageUrl(site.getIconUrl(), size, size, PhotonUtils.Quality.HIGH,
                 site.isPrivateWPComAtomic());
+    }
+
+    public static SiteAccessibilityInfo getAccessibilityInfoFromSite(@NotNull SiteModel site) {
+        SiteVisibility siteVisibility;
+
+        if (site.isPrivateWPComAtomic()) {
+            siteVisibility = SiteVisibility.PRIVATE_ATOMIC;
+        } else if (site.isPrivate()) {
+            siteVisibility = SiteVisibility.PRIVATE;
+        } else {
+            siteVisibility = SiteVisibility.PUBLIC;
+        }
+
+        return new SiteAccessibilityInfo(siteVisibility, isPhotonCapable(site));
     }
 
     public static ArrayList<Integer> getCurrentSiteIds(SiteStore siteStore, boolean selfhostedOnly) {

--- a/WordPress/src/main/java/org/wordpress/android/util/SiteUtilsWrapper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/util/SiteUtilsWrapper.kt
@@ -1,7 +1,9 @@
 package org.wordpress.android.util
 
 import dagger.Reusable
+import org.jetbrains.annotations.NotNull
 import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.reader.utils.SiteAccessibilityInfo
 import javax.inject.Inject
 
 /**
@@ -14,4 +16,6 @@ import javax.inject.Inject
 @Reusable
 class SiteUtilsWrapper @Inject constructor() {
     fun isPhotonCapable(site: SiteModel): Boolean = SiteUtils.isPhotonCapable(site)
+    fun getAccessibilityInfoFromSite(site: SiteModel): SiteAccessibilityInfo
+            = SiteUtils.getAccessibilityInfoFromSite(site)
 }

--- a/WordPress/src/main/java/org/wordpress/android/util/SiteUtilsWrapper.kt
+++ b/WordPress/src/main/java/org/wordpress/android/util/SiteUtilsWrapper.kt
@@ -1,7 +1,6 @@
 package org.wordpress.android.util
 
 import dagger.Reusable
-import org.jetbrains.annotations.NotNull
 import org.wordpress.android.fluxc.model.SiteModel
 import org.wordpress.android.ui.reader.utils.SiteAccessibilityInfo
 import javax.inject.Inject
@@ -16,6 +15,6 @@ import javax.inject.Inject
 @Reusable
 class SiteUtilsWrapper @Inject constructor() {
     fun isPhotonCapable(site: SiteModel): Boolean = SiteUtils.isPhotonCapable(site)
-    fun getAccessibilityInfoFromSite(site: SiteModel): SiteAccessibilityInfo
-            = SiteUtils.getAccessibilityInfoFromSite(site)
+    fun getAccessibilityInfoFromSite(site: SiteModel): SiteAccessibilityInfo =
+            SiteUtils.getAccessibilityInfoFromSite(site)
 }

--- a/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt
+++ b/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt
@@ -395,8 +395,7 @@ class PostListViewModel @Inject constructor(
                     featuredImageUrl,
                     photonWidth,
                     photonHeight,
-                    !SiteUtils.isPhotonCapable(connector.site),
-                    connector.site.isPrivateWPComAtomic
+                    SiteUtils.getAccessibilityInfoFromSite(connector.site)
             )
 
     fun updateAuthorFilterIfNotSearch(authorFilterSelection: AuthorFilterSelection): Boolean {

--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/FeaturedImageHelperTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/FeaturedImageHelperTest.kt
@@ -14,7 +14,6 @@ import org.assertj.core.api.Assertions.assertThat
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.ArgumentMatchers.anyBoolean
 import org.mockito.ArgumentMatchers.anyInt
 import org.mockito.ArgumentMatchers.anyLong
 import org.mockito.junit.MockitoJUnitRunner

--- a/WordPress/src/test/java/org/wordpress/android/ui/posts/FeaturedImageHelperTest.kt
+++ b/WordPress/src/test/java/org/wordpress/android/ui/posts/FeaturedImageHelperTest.kt
@@ -1,6 +1,7 @@
 package org.wordpress.android.ui.posts
 
 import com.nhaarman.mockitokotlin2.KArgumentCaptor
+import com.nhaarman.mockitokotlin2.any
 import com.nhaarman.mockitokotlin2.anyOrNull
 import com.nhaarman.mockitokotlin2.argThat
 import com.nhaarman.mockitokotlin2.argumentCaptor
@@ -29,6 +30,7 @@ import org.wordpress.android.fluxc.store.UploadStore
 import org.wordpress.android.ui.posts.FeaturedImageHelper.EnqueueFeaturedImageResult
 import org.wordpress.android.ui.posts.FeaturedImageHelper.FeaturedImageState
 import org.wordpress.android.ui.reader.utils.ReaderUtilsWrapper
+import org.wordpress.android.ui.reader.utils.SiteAccessibilityInfo
 import org.wordpress.android.ui.uploads.UploadServiceFacade
 import org.wordpress.android.util.FluxCUtilsWrapper
 import org.wordpress.android.util.SiteUtilsWrapper
@@ -43,12 +45,14 @@ class FeaturedImageHelperTest {
     private val readerUtilsWrapper: ReaderUtilsWrapper = mock()
     private val fluxCUtilsWrapper: FluxCUtilsWrapper = mock()
     private val siteUtilsWrapper: SiteUtilsWrapper = mock()
+    private val siteAccessibilityInfo: SiteAccessibilityInfo = mock()
     private val dispatcher: Dispatcher = mock()
 
     private lateinit var featuredImageHelper: FeaturedImageHelper
 
     @Before
     fun setUp() {
+        whenever(siteUtilsWrapper.getAccessibilityInfoFromSite(any())).thenReturn(siteAccessibilityInfo)
         featuredImageHelper = FeaturedImageHelper(
                 uploadStore,
                 mediaStore,
@@ -318,8 +322,7 @@ class FeaturedImageHelperTest {
                 eq("https://testing.com/url.jpg"),
                 anyInt(),
                 anyInt(),
-                anyBoolean(),
-                anyBoolean()
+                eq(siteAccessibilityInfo)
         )
     }
 
@@ -344,8 +347,7 @@ class FeaturedImageHelperTest {
                 "https://testing.com/thumbnail.jpg"),
                 anyInt(),
                 anyInt(),
-                anyBoolean(),
-                anyBoolean()
+                eq(siteAccessibilityInfo)
         )
     }
 

